src/lib/homestore_backend/snapshot_receive_handler.cpp

```diff
@@ -59,13 +58,14 @@ int HSHomeObject::SnapshotReceiveHandler::process_shard_snapshot_data(ResyncShar
     shard_sb.chunk_id = chunk_id;
 
     homestore::MultiBlkId blk_id;
-    const auto hints = home_obj_.chunk_selector()->chunk_to_hints(chunk_id);
+    const auto hints = home_obj_.chunk_selector()->chunk_to_hints(shard_sb.chunk_id);
     auto status = homestore::data_service().alloc_blks(
         sisl::round_up(sizeof(shard_sb), homestore::data_service().get_blk_size()), hints, blk_id);
     if (status != homestore::BlkAllocStatus::SUCCESS) {
         LOGE("Failed to allocate blocks for shard {}", shard_meta.shard_id());
```

Comment:
not sure will we always baseline resync to a brand new member? if not, maybe we need to let emergency gc kick in here